crates/sql-sink/src/bind.rs

```diff
@@ -0,0 +1,79 @@
+use std::str::FromStr;
+
+use rust_decimal::Decimal;
+use sqlx::database::HasArguments;
+use sqlx::postgres::PgArguments;
+use sqlx::query::Query;
+use sqlx::sqlite::SqliteArguments;
+use sqlx::{Database, Postgres, Sqlite};
+
+use fluvio_model_sql::{Type, Value};
+
+use crate::db::Db;
+
+pub const NAIVE_DATE_TIME_FORMAT: &str = "%Y-%m-%d %H:%M:%S%.f";
+
+pub trait Bind<DB: Database> {
+    fn bind_value<'a>(
+        query: Query<'a, DB, <DB as HasArguments<'a>>::Arguments>,
+        value: &Value,
+    ) -> anyhow::Result<Query<'a, DB, <DB as HasArguments<'a>>::Arguments>>;
+}
+
+impl Bind<Postgres> for Db {
+    fn bind_value<'a>(
+        query: Query<'a, Postgres, PgArguments>,
+        value: &Value,
+    ) -> anyhow::Result<Query<'a, Postgres, PgArguments>> {
+        let query = match value.type_ {
+            Type::Bool => query.bind(bool::from_str(&value.raw_value)?),
+            Type::Char => query.bind(i8::from_str(&value.raw_value)?),
+            Type::SmallInt => query.bind(i16::from_str(&value.raw_value)?),
+            Type::Int => query.bind(i32::from_str(&value.raw_value)?),
+            Type::BigInt => query.bind(i64::from_str(&value.raw_value)?),
+            Type::Float => query.bind(f32::from_str(&value.raw_value)?),
+            Type::DoublePrecision => query.bind(f64::from_str(&value.raw_value)?),
+            Type::Text => query.bind(value.raw_value.clone()),
```

Comment:
If it is not too complicated, can we make `value` owned and avoid cloning here? It will affect all the chain up to `Operation` but it will be more efficient.